// Copyright (c) 2017-2019, NVIDIA CORPORATION.  All rights reserved.

/**
* postprocessing_config.proto: Protocol buffer definition for configuring the post-processing
* block of the DetectNet-v2 model.
*/

syntax = "proto3";

message DBSCANConfig{
    // DBSCAN parameters
    float dbscan_eps = 1;
    float dbscan_min_samples = 2;
    int32 neighborhood_size = 3;
    float dbscan_confidence_threshold = 4;
}

message ClusteringConfig {
    // Grid cells with coverage lower than this threshold will be ignored
    float coverage_threshold = 1;
    int32 minimum_bounding_box_height = 2;
    DBSCANConfig dbscan_config = 3;
    message BboxColor {
        int32 R = 1;
        int32 G = 2;
        int32 B = 3;
    }
    BboxColor bbox_color = 4;
}

// Create a dictionary of post processing config.
message PostprocessingConfig {
    // Post processing config.
    map <string, ClusteringConfig> classwise_clustering_config = 1;
    int32 linewidth = 2;
    int32 stride = 3;
}

// Create a dictionary of post processing config for CenterPose.
message CenterPoseConfig {
    // CenterPose post processing config, intrinsic matrix.
    float visualization_threshold = 1;
    float principle_point_x = 2;
    float principle_point_y = 3;
    float focal_length_x = 4;
    float focal_length_y = 5;
    float skew = 6;
    // CenterPose visualization parameters.
    float axis_size = 7;
    int32 square_size = 8;
    int32 line_weight = 9;
    float scale_text = 10;
}